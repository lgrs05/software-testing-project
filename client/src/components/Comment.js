import Avatar from "@material-ui/core/Avatar";
import ListItem from "@material-ui/core/ListItem";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import ListItemText from "@material-ui/core/ListItemText";
import PropTypes from 'prop-types';
import React from "react";
import Typography from "@material-ui/core/Typography";

const avatarSrc = "https://source.unsplash.com/2ShvY8Lf6l0/800x599";

function Comment({name, message, time}) {
    return (
        <ListItem alignItems="flex-start">
            <ListItemAvatar>
                <Avatar alt={name} src= {avatarSrc} />
            </ListItemAvatar>
            <ListItemText
                primary={name}
                secondary={
                    <React.Fragment>
                        <Typography component="span" inline color="textPrimary">
                            {message}
                        </Typography>
                        <br />{time}
                    </React.Fragment>
                }
            />
        </ListItem>
    );
}


Comment.propTypes = {
    name    : PropTypes.string.isRequired,
    message : PropTypes.string.isRequired,
    time    : PropTypes.string.isRequired
};


export default (Comment);