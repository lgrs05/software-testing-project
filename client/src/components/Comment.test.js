import Comment from './Comment';
import React from 'react';
import renderer from 'react-test-renderer';

let comm = {};

beforeEach(() => {
    comm = {
        name    : 'Pedro',
        message : "This is a good message",
        time    : new Date(2019, 2, 21, 12, 0, 0, 0).toLocaleString()
    };
});

it('renders correctly', () => {
    const tree = renderer.create(<Comment
        name={comm.name}
        message={comm.message}
        time={comm.time} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
