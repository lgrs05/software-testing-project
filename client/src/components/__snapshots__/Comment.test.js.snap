// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="MuiListItem-root-1 MuiListItem-default-4 MuiListItem-dense-5 MuiListItem-gutters-9 MuiListItem-alignItemsFlexStart-6"
  disabled={false}
>
  <div
    className="MuiAvatar-root-16 MuiListItemAvatar-alignItemsFlexStart-14"
  >
    <img
      alt="Pedro"
      className="MuiAvatar-img-18"
      src="https://source.unsplash.com/2ShvY8Lf6l0/800x599"
    />
  </div>
  <div
    className="MuiListItemText-root-19"
  >
    <span
      className="MuiTypography-root-25 MuiTypography-subheading-32 MuiListItemText-primary-22"
    >
      Pedro
    </span>
    <p
      className="MuiTypography-root-25 MuiTypography-body1-34 MuiTypography-colorTextSecondary-58 MuiListItemText-secondary-23"
    >
      <span
        className="MuiTypography-root-25 MuiTypography-body1-34 MuiTypography-colorTextPrimary-57 MuiTypography-inline-60"
      >
        This is a good message
      </span>
      <br />
      2019-3-21 12:00:00
    </p>
  </div>
</li>
`;
